scripts/release-checks/assert-valid-peer-dependency.mts

```diff
@@ -0,0 +1,54 @@
+import { Log, PackageJson, ReleasePrecheckError } from '@angular/ng-dev';
+import path from 'path';
+import url from 'url';
+import fs from 'fs';
+import semver from 'semver';
+
+interface PackageJson {
+  name: string;
+  peerDependencies: Record<string, string>;
+}
+
+/** Path to the current directory. */
+const currentDir = path.dirname(url.fileURLToPath(import.meta.url));
+
+const projectDir = path.join(currentDir, '../../');
+
+/**
+ * Ensures that the Angular framework peer dependency has been correctly updated
+ * to support given Angular versions. The following rules apply:
+ *
+ *    - TODO
+ */
+export async function assertValidFrameworkPeerDependency(newVersion: semver.SemVer) {
+  const major = newVersion.major;
+  const isPrerelease = !!newVersion.prerelease[0];
+  const isMajor = newVersion.minor === 0;
+
+  const ngDevkitBuildAngularPkgJson = await loadPackageJson(
+    path.join(projectDir, 'packages/angular_devkit/build_angular/package.json'),
+  );
+
+  let expectedFwDep: string;
+  if (isMajor) {
+    if (isPrerelease) {
+      expectedFwDep = `^${major}.0.0-next.0`;
+    } else {
+      expectedFwDep = `^${major}.0.0 || ^${major}.${newVersion.minor + 1}.0-next.0`;
```

Comment:
Do we normally include `-next` in non-prerelease versions? For example, I thought CLI `15.2.0` would only have a peer dep on FW `^15.0.0`?

Reply:
Correct on a stable version we do not add add any pre-release support.